bat-utils/lib/hapi-rate-limiter.js

```diff
@@ -0,0 +1,149 @@
+const boom = require('@hapi/boom')
+const Netmask = require('netmask').Netmask
+const {
+  RateLimiterRedis
+} = require('rate-limiter-flexible')
+const _ = require('underscore')
+const underscore = _
+
+const braveHapi = require('./extras-hapi')
+const whitelist = require('./hapi-auth-whitelist')
+
+const pluginName = 'rateLimitRedisPlugin'
+
+const graylist = {
+  addresses: process.env.IP_GRAYLIST && process.env.IP_GRAYLIST.split(',')
+}
+
+if (graylist.addresses) {
+  graylist.authorizedAddrs = []
+  graylist.authorizedBlocks = []
+
+  graylist.addresses.forEach((entry) => {
+    if ((entry.indexOf('/') === -1) && (entry.split('.').length === 4)) return graylist.authorizedAddrs.push(entry)
```

Comment:
if you're trying to validate an ipv4 address here, you can just use https://www.npmjs.com/package/validate-ip-node which is already a dep of this library

Reply:
there's a lot of places where we can / should clean up code, but for this PR I asked Michael to focus on making the existing code work with the latest hapi with as few changes to make it easier to verify. for something like this I'd prefer we open an issue and address in a follow up PR

Reply:
https://github.com/brave-intl/bat-ledger/issues/743